apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceMiddleware.cs

```diff
@@ -29,24 +29,36 @@ namespace Google.Cloud.Diagnostics.AspNetCore
     internal sealed class CloudTraceMiddleware
     {
         private readonly RequestDelegate _next;
+        private readonly IManagedTracerFactory _tracerFactory;
+        private readonly IHttpContextAccessor _accessor;
 
         /// <summary>
         /// Create a new instance of <see cref="CloudTraceMiddleware"/>.
         /// </summary>
         /// <param name="next">The next request delegate. Cannot be null.</param>
-        public CloudTraceMiddleware(RequestDelegate next)
+        /// <param name="tracerFactory">A factory to create <see cref="IManagedTracer"/>s.</param>
+        /// <param name="accessor">Access to <see cref="HttpContext"/>.</param>
+        public CloudTraceMiddleware(
+            RequestDelegate next, IManagedTracerFactory tracerFactory, IHttpContextAccessor accessor)
         {
             _next = GaxPreconditions.CheckNotNull(next, nameof(next));
+            _tracerFactory = GaxPreconditions.CheckNotNull(tracerFactory, nameof(tracerFactory));
+            _accessor = GaxPreconditions.CheckNotNull(accessor, nameof(accessor));
         }
 
         /// <summary>
-        /// Invokes the next <see cref="RequestDelegate"/> and trace the time 
+        /// Invokes the next <see cref="RequestDelegate"/> and traces the time 
         /// taken for the next delegate to run, reporting the results to the
         /// Stackdriver Trace API.
         /// </summary>
-        public async Task Invoke(HttpContext httpContext, IManagedTracer tracer)
+        public async Task Invoke(HttpContext httpContext, TraceHeaderContext traceHeaderContext)
```

Comment:
Might be good to add a \<param\> tag for this, which also indicates that it cannot be null.